### PR TITLE
Add fire barrels

### LINKED ISSUE #30515: Furniture - fire barrel
**Is your feature request related to a problem? Please describe.**  
metal barrels are a great way to safely contain fires, but can't be used for this in game.

**Describe the solution you'd like**  
Make metal barrels a placeable furniture via the construction menu, and function like a brazier.

**Describe alternatives you've considered**  
Could also be (a)ctivated like folding furniture, as with braziers.

**Additional context**  
This is a simple thing to add. If you're thinking of getting into contributing, this would be a great starting point.

image

### PR DESCRIPTION
**Pull request by tinukedaya**
#### Summary
```SUMMARY: Content "Add fire barrels"```

#### Purpose of change
Add another option to contain fire.
Closes #30515

#### Describe the solution
JSON editing. Used the respective drums for stats, but removed all the containing, boiling tags as it has holes in it. 

#### Additional context
Trying to work with github desktop. Hope everything works fine.

### CODE REVIEW

File: data/json/recipes/recipe_others.json [hunk 1]
```diff
@@ -1728,6 +1728,32 @@
...
+    "type": "recipe",
+    "result": "55gal_firebarrel",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "survival",
+    "difficulty": 1,
+    "time": 2000,
+    "autolearn": true,
+    "qualities": [ { "id": "HAMMER", "level": 1 } ],
+    "tools": [ [ [ "spike", -1], [ "sharp_rock", -1], [ "rebar", -1], [ "spear_rebar", -1] ] ],
```
Comment (AMurkin) [inline data/json/recipes/recipe_others.json]:
```suggestion
    "tools": [ [ [ "spike", -1 ], [ "sharp_rock", -1 ], [ "rebar", -1 ], [ "spear_rebar", -1 ] ] ],
```

File: data/json/recipes/recipe_others.json [hunk 2]
```diff
@@ -1728,6 +1728,32 @@
...
+    "type": "recipe",
+    "result": "30gal_firebarrel",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "survival",
+    "difficulty": 1,
+    "time": 2000,
+    "autolearn": true,
+    "qualities": [ { "id": "HAMMER", "level": 1 } ],
+    "tools": [ [ [ "spike", -1], [ "sharp_rock", -1], [ "rebar", -1], [ "spear_rebar", -1] ] ],
```
Comment (AMurkin) [inline data/json/recipes/recipe_others.json]:
```suggestion
    "tools": [ [ [ "spike", -1 ], [ "sharp_rock", -1 ], [ "rebar", -1 ], [ "spear_rebar", -1 ] ] ],
```

File: data/json/furniture.json [hunk 3]
```diff
@@ -3774,6 +3774,57 @@
...
+    "flags": [ "PLACE_ITEM", "TRANSPARENT", "FIRE_CONTAINER", "EASY_DECONSTRUCT" ],
+    "deployed_item": "30gal_firebarrel",
+    "examine_action": "fireplace",
+    "max_volume": 4000,
+    "bash": {
+      "str_min": 8,
+      "str_max": 30,
+      "sound": "metal screeching!",
+      "sound_fail": "clang!",
+      "items": [
```
Comment (AMurkin) [inline data/json/furniture.json]: Should be one line:
`"items": [ { "item": "scrap", "count": [ 5, 15 ] }, { "item": "sheet_metal_small", "count": [ 1, 9 ] } ]`

Comment (tinukedaya) [reply data/json/furniture.json]: Alright I'll do that. But all the other recipes have it this way. I really just copy pasted the other recipe. Wouldn't it be a problem that my recipes are formatted differently?

Comment (AMurkin) [reply data/json/furniture.json]: That's the last change that JSON-formatter wants.

File: data/json/furniture.json [hunk 4]
```diff
@@ -3774,6 +3774,54 @@
       ]
     }
   },
+  {
+    "type": "furniture",
```
Comment (I-am-Erk) [inline data/json/furniture.json]: Don't forget to add your description strings from the items to the furnitures as well.

Comment (tinukedaya) [reply data/json/furniture.json]: Ooops, definitely would have forgot that.

File: data/json/furniture.json [hunk 5]
```diff
@@ -3774,6 +3774,56 @@
...
+    "id": "f_55gal_firebarrel",
+    "name": "fire barrel (200L)",
+    "description": "A large metal barrel used to contain a fire.  It has multiple holes punched in it's walls for air supply.  Fires set in a fire barrel will not spread to surrounding flammable objects.",
```
Comment (matskuman5) [inline data/json/furniture.json]:
```suggestion
    "description": "A large metal barrel used to contain a fire.  It has multiple holes punched in its walls for air supply.  Fires set in a fire barrel will not spread to surrounding flammable objects.",
```

File: data/json/furniture.json [hunk 6]
```diff
@@ -3774,6 +3774,56 @@
...
+        { "item": "sheet_metal_small", "count": [ 3, 10 ] },
+        { "item": "sheet_metal", "count": [ 0, 1 ] }
+      ]
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_30gal_firebarrel",
+    "name": "fire barrel (100L)",
+    "description": "A large metal barrel used to contain a fire.  It has multiple holes punched in it's walls for air supply.  Fires set in a fire barrel will not spread to surrounding flammable objects.",
```
Comment (matskuman5) [inline data/json/furniture.json]:
```suggestion
    "description": "A large metal barrel used to contain a fire.  It has multiple holes punched in its walls for air supply.  Fires set in a fire barrel will not spread to surrounding flammable objects.",
```

File: data/json/items/tool/deployable.json [hunk 7]
```diff
@@ -15,6 +15,40 @@
     "use_action": { "type": "deploy_furn", "furn_type": "f_brazier" },
     "qualities": [ [ "COOK", 1 ], [ "BOIL", 2 ], [ "CONTAIN", 1 ] ]
   },
+  {
+    "id": "55gal_firebarrel",
+    "type": "TOOL",
+    "name": "fire barrel (200L)",
+    "name_plural": "fire barrels (200L)",
+    "description": "A large metal barrel used to contain a fire.  It has multiple holes punched in it's walls for air supply.  Fires set in a fire barrel will not spread to surrounding flammable objects.",
```
Comment (matskuman5) [inline data/json/items/tool/deployable.json]:
```suggestion
    "description": "A large metal barrel used to contain a fire.  It has multiple holes punched in its walls for air supply.  Fires set in a fire barrel will not spread to surrounding flammable objects.",
```

File: data/json/items/tool/deployable.json [hunk 8]
```diff
@@ -15,6 +15,40 @@
...
+    "symbol": "0",
+    "color": "light_gray",
+    "use_action": { "type": "deploy_furn", "furn_type": "f_55gal_firebarrel" }
+  },
+  {
+    "id": "30gal_firebarrel",
+    "type": "TOOL",
+    "name": "fire barrel (100L)",
+    "name_plural": "fire barrels (100L)",
+    "description": "A large metal barrel used to contain a fire.  It has multiple holes punched in it's walls for air supply.  Fires set in a fire barrel will not spread to surrounding flammable objects.",
```
Comment (matskuman5) [inline data/json/items/tool/deployable.json]:
```suggestion
    "description": "A large metal barrel used to contain a fire.  It has multiple holes punched in its walls for air supply.  Fires set in a fire barrel will not spread to surrounding flammable objects.",
```

**Pull request merged**